### ISSUE TITLE
Bug: Unparsable segment "UNION ALL" in "WITH"

### ISSUE THREAD

**Issue by mrshu** [alanmcruickshank/sqlfluff#162]
Running `sqlfluff lint` (`master` branch) on the following SQL file

```sql
WITH result AS (
    SELECT
        customer
    FROM sales_eu AS s
    UNION ALL
    SELECT
        customer
    FROM sales_us AS s2
)

SELECT * FORM result
```
Results in the following:

```
L:   5 | P:   5 |  PRS | Found unparsable segment @L005P005: 'UNION ALL\n    SELECT...'
L:   7 | P:   9 | L003 | Indentation not consistent with line #6
L:  11 | P:  10 |  PRS | Found unparsable segment @L011P010: 'FORM result...'
```

I am not sure whether it is the intention of `sqlfluff` not to support `WITH` statements, but since they are in the standard since [`SQL:1999`](https://modern-sql.com/feature/with),  I believe a query like this one should at least get parsed.

I am happy to help with debugging this further!

**Comment by alanmcruickshank** [comment]
Good catch. PR coming live shortly.
